spec/policy/headers/headers_spec.lua

```diff
@@ -0,0 +1,204 @@
+local HeadersPolicy = require('apicast.policy.headers')
+
+describe('Headers policy', function()
+  -- Apply the operations to the same header in all the tests for simplicity
+  local header = 'test_header'
+
+  describe('.rewrite', function()
+    local header_type = 'request' -- rewrite() only modifies request headers
+
+    -- ngx.req.get_headers and ngx.set_headers are not available in these tests
+    -- so we need to mock them.
+    -- Note: ngx.req_headers()['some_header'] can return a single value or a
+    -- table.
+    local ngx_req_headers = {}
+    ngx.req = {
```

Comment:
Hm. This is going to override it locally. Would it be possible to have this done by `stub` in some before_each hook? I think this will leak through tests.

Reply:
You are right. What's defined here also applies to the `describe(.header_filter')` block below. Which is fine in this case, but I can see how this can create problems in other cases.

Reply:
Moved the stubs to a `before_each` clause so what's defined there does not leak to other tests.